### DISCUSSION TITLE
Fireball Collision not Firing

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#2003] [Help]
I'm currently using [MinestomPvP](https://github.com/JasperTheMinecraftDev/MinestomPvP), and I've implemented a `Fireball` class as follows: 

```java
public class Fireball extends CustomEntityProjectile implements ItemHoldingProjectile {

    public Fireball(@Nullable Entity shooter) {
        super(shooter, EntityType.FIREBALL, true);
    }

    @Override
    public void onHit(Entity entity) {
        System.out.println("fireball hit");
        triggerStatus((byte) 3);

        remove();
    }

    @Override
    public void onStuck() {
        remove();
    }

    @Override
    public void tick(long time) {
        super.tick(time);
    }

    @Override
    public void setItem(@NotNull ItemStack item) {
        ((FireballMeta) getEntityMeta()).setItem(item);
    }
}
```

and I later spawn/fire this via

```java
Fireball fireball = new Fireball(event.getPlayer());
Pos position = event.getPlayer().getPosition();
Vec direction = event.getPlayer().getPosition().direction();
fireball.setVelocity(direction.normalize().mul(20));
fireball.setGravity(0.0, 0.0);
fireball.setInstance(event.getPlayer().getInstance(), position.add(0, event.getPlayer().getEyeHeight(),0));
```

however it seems to not fire `ProjectileHitEvent.ProjectileEntityHitEvent.class` when it collides with an Entity

**Comment by GreatWyrm** [comment]
What class does CustomEntityProjectile inherit from? EntityProjectile?
Also, we don't have a ProjectileEntityHitEvent, it's ProjectileCollideWithEntityEvent.

**Reply by para-dise** [reply]
`CustomEntityProjectile` is defined as `public class CustomEntityProjectile extends Entity`, and `ProjectileEntityHitEvent` is a MinestomPvP class (see [here)](https://github.com/JasperTheMinecraftDev/MinestomPvP/blob/master/src/main/java/io/github/bloepiloepi/pvp/events/ProjectileHitEvent.java)

**Reply by GreatWyrm** [reply]
Ah, okay, I'm not super familiar with that library, but my guess is that something with the projectile is breaking because you're not calling the `shoot()` method. Try that?

[Here's](https://github.com/JasperTheMinecraftDev/MinestomPvP/blob/master/src/main/java/io/github/bloepiloepi/pvp/projectile/ProjectileListener.java#L148) the method I'm looking at in the library.
